setup.py

```diff
@@ -43,17 +43,17 @@
 setup(
     author="Sarah-Claude Bourdeau-Goulet",
     author_email="bourdeau-goulet.sarah-claude@ouranos.ca",
-    python_requires=">=3.7",
+    python_requires=">=3.8",
     classifiers=[
         "Development Status :: 2 - Pre-Alpha",
         "Intended Audience :: Developers",
         "License :: OSI Approved :: Apache Software License",
         "Natural Language :: English",
         "Programming Language :: Python :: 3",
-        "Programming Language :: Python :: 3.7",
         "Programming Language :: Python :: 3.8",
         "Programming Language :: Python :: 3.9",
         "Programming Language :: Python :: 3.10",
+        "Programming Language :: Python :: 3.11",
```

Comment:
I haven't tried installing all the deps for Python3.11. Is CLISOPS the blocking factor here as well?

Reply:
i don't know why it's <3.10. I think @sarahclaude had issues previously and pinned it. She's trying again without the pin.

Reply:
I had issues with CLISOPS with 3.11, I don't exactly remeber if I had other issues in spirograph but I could try running it again with 3.11 to test.

Reply:
It works with 3.11, you can remove python =<3.10.
I only get this warning; NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @jit

Reply:
@sarahclaude I wouldn't remove the pin on python in the environment, since it helps reduce the time needed to solve the environment on ReadTheDocs. Generally, pinning against the lowest supported version is a guarantee that the documentation will build (it doesn't need to be cutting edge).

The other error concerning numba is coming from xclim and will be fixed in the next release (possibly today!).

Reply:
Do you mean, it's better to leave python >=3.10 instead of no pins or equal to 3.11?

Reply:
It's better to leave a loose-ish pin on one release version (i.e. `python >=3.10,<3.11`) since it allows for patch updates to Python (3.10.1/2/3/4/...). We don't want mamba on ReadTheDocs to have to determine which version of Python to take when it builds the docs, as that creates a bunch of unnecessary compute cycles.

The Python version needed to install the library is listed in the `setup.py` so if we pip install in any old environment, so long as the Python version matches there, no issues will be had.

Reply:
Thanks for the info!